cibuildwheel/macos.py

```diff
@@ -191,39 +193,39 @@ def build(options: BuildOptions) -> None:
             call(before_build_prepared, env=env, shell=True)
 
         # build the wheel
-        if os.path.exists(built_wheel_dir):
+        if built_wheel_dir.exists():
             shutil.rmtree(built_wheel_dir)
-        os.makedirs(built_wheel_dir)
-        # os.path.abspath is need. Without it pip wheel may try to fetch package from pypi.org
+        built_wheel_dir.mkdir(parents=True)
+        # Path.resolve() is needed. Without it pip wheel may try to fetch package from pypi.org
         # see https://github.com/joerick/cibuildwheel/pull/369
-        call(['pip', 'wheel', os.path.abspath(options.package_dir), '-w', built_wheel_dir, '--no-deps'] + get_build_verbosity_extra_flags(options.build_verbosity), env=env)
-        built_wheel = glob(os.path.join(built_wheel_dir, '*.whl'))[0]
+        call(['pip', 'wheel', str(options.package_dir.resolve()), '-w', str(built_wheel_dir), '--no-deps'] + get_build_verbosity_extra_flags(options.build_verbosity), env=env)
+        built_wheel = next(built_wheel_dir.glob('*.whl'))
 
         # repair the wheel
-        if os.path.exists(repaired_wheel_dir):
+        if repaired_wheel_dir.exists():
             shutil.rmtree(repaired_wheel_dir)
-        os.makedirs(repaired_wheel_dir)
-        if built_wheel.endswith('none-any.whl') or not options.repair_command:
+        repaired_wheel_dir.mkdir(parents=True)
+        if built_wheel.name.endswith('none-any.whl') or not options.repair_command:
             # pure Python wheel or empty repair command
-            shutil.move(built_wheel, repaired_wheel_dir)
+            shutil.move(str(built_wheel), repaired_wheel_dir)
```

Comment:
Hmmm, `shutil.move` is slightly smarter (just looked it up: apparently it works when the files are on different filesystems), and I suppose `/tmp` could be on a different partition?

@joerick, do you remember why `shutil.move` rather than `os.rename`? Is it worth trying `os.rename`/`Path.rename`?

Reply:
This string conversion here is needed?

Reply:
@joerick made the same comment already. mypy does something weird:

```
cibuildwheel/macos.py:210: error: Argument 1 to "move" has incompatible type "Path"; expected "str"
```

I think it's wrong, but to silence the error, we probably need `str(...)`

Reply:
...
Some bug report to mypy?

Reply:
Turns out it's not wrong... exactly

https://bugs.python.org/issue32689
https://github.com/python/typeshed/pull/3559

Reply:
> @joerick, do you remember why `shutil.move` rather than `os.rename`? Is it worth trying `os.rename`/`Path.rename`?

Probably just because there are fewer gotchas, as you mention. But Path.rename is worth a shot.

Reply:
> Turns out it's not wrong... exactly
> 
> https://bugs.python.org/issue32689
> [python/typeshed#3559](https://github.com/python/typeshed/pull/3559)

Ah, thanks for finding that!

> Probably just because there are fewer gotchas, as you mention. But Path.rename is worth a shot.

OK, just did that. I think I just figured out á potential reason for it: the second argument to `shutil.move` (destination) was in some cases (like this one) a directory. I suppose this wouldn't work with `rename`.